tests/plugins/textwatcher/manual/throttle.html

```diff
@@ -0,0 +1,37 @@
+<style>
+#logger {
+	height: 200px;
+	overflow-y: auto;
+}
+</style>
+<strong>Logger: </strong>
+<ul id="logger"></ul>
+<div id="editor1" ></div>
+
+<script>
+	var logger = document.getElementById( 'logger' ),
+		editor = CKEDITOR.replace( 'editor1', {
+		on: {
+			instanceReady: function() {
+				var counter = 0,
+					counterInterval = 100,
+
+					textWatcher = new CKEDITOR.plugins.textWatcher( editor, function( selectionRange ) {
+					var matchEl = document.createElement( 'li' );
+
+					matchEl.innerText = 'Time: ' + counter + 'ms, Text: ' + selectionRange.startContainer.getText();
+					logger.appendChild( matchEl );
+
+					if ( counter == 0 ) {
+						setInterval( function() {
```

Comment:
That's a bit overcomplicated solution… use simply `performance.now()` and display a difference between last and current call.

*However* currently `performance.now()` has an issue that it might be off by +/- 2 ms, see [mdn for more details](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now):

> The timestamp is not actually high-resolution. To mitigate security threats such as Spectre, browsers currently round the result to varying degrees. (Firefox started rounding to 2 milliseconds in Firefox 59.) Some browsers may also slightly randomize the timestamp. The precision may improve again in future releases; browser developers are still investigating these timing attacks and how best to mitigate them.

In that case I propose to set throttling in the sample to 2002 ms, and issue description should still say that the calls should be no more often than 2000 ms.

Finally make sure to put a short comment (even simply to this comment) in the code, right above the 2002 threshold value.

Reply:
After doing that you'll be able to remove this fragment:

> Other details Measured time by logger can have slight error +/- 100ms.